hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/ServiceFactory.java

```diff
@@ -107,7 +107,7 @@ private ServiceFactory(
      */
     @Nonnull
     public static <S> ServiceFactory<S> withCreateFn(
-            @Nonnull FunctionEx<JetInstance, ? extends S> createServiceFn
+            @Nonnull FunctionEx<Context, ? extends S> createServiceFn
```

Comment:
We should use `? super Context` here. `? super ProcessorSupplier.Contex` would be even better. Same in `SinkBuilder.sinkBuilder()`